### DISCUSSION TITLE
Improve asr (whisper) with speaker diarization result

### DISCUSSION THREAD

**Discussion by yinruiqing** [pyannote/pyannote-audio#1144] [Ideas]
I combined whisper and pyannote.audio in [pyannote-whisper](https://github.com/yinruiqing/pyannote-whisper) to get "who speaks when and what" from an audio file. The strategy is very simple: choose the major speaker in a sentence. The performance may degrade a lot for fast speech turns. Can we find a more elegant way to combine these tools? For example, do diarization first and then feed to whisper (this may also cause some problems in the overlap part).

**Comment by hbredin** [comment]
I tried this [here](https://gist.github.com/hbredin/049f2b629700bcea71324d2c1e7f8337).

One way to improve it would be to merge adjacent speech turns of the same speaker so that Whisper can benefit from more context.

This does not solve the overlapping speech issue, however.
